src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1187,8 +1235,11 @@ private async Task ProcessCompilationEventsAsync(AnalysisScope analysisScope, An
                     // Kick off tasks to execute syntax tree actions.
                     var syntaxTreeActionsTask = Task.Run(() => ExecuteSyntaxTreeActions(analysisScope, analysisStateOpt, cancellationToken), cancellationToken);
 
+                    // Kick off tasks to execute additional file actions.
+                    var additionalFileActionsTask = Task.Run(() => ExecuteAdditionalFileActions(analysisScope, analysisStateOpt, cancellationToken), cancellationToken);
+
                     // Wait for all worker threads to complete processing events.
-                    await Task.WhenAll(workerTasks.Concat(syntaxTreeActionsTask)).ConfigureAwait(false);
+                    await Task.WhenAll(workerTasks.Concat(syntaxTreeActionsTask).Concat(additionalFileActionsTask)).ConfigureAwait(false);
```

Comment:
>.Concat(syntaxTreeActionsTask).Concat(additionalFileActionsTask) [](start = 50, length = 64)

Consider allocating the `workerTasks` array larger and explicitly adding to the array rather than using `Concat()`.

Reply:
`workerTasks` array is `Task<CompilationCompletedEvent>[]`, while these `syntaxTreeActionsTask` and `additionalFileActionsTask` return `Task`.